java/vector/src/test/java/org/apache/arrow/vector/complex/impl/TestPromotableWriter.java

```diff
@@ -92,6 +93,19 @@ public void testPromoteToUnion() throws Exception {
 
       assertFalse("4 shouldn't be null", accessor.isNull(4));
       assertEquals(100, accessor.getObject(4));
+
+      container.clear();
+      container.allocateNew();
+
+      ComplexWriterImpl newWriter = new ComplexWriterImpl(EMPTY_SCHEMA_PATH, container);
+
+      writer.start();
+
+      writer.setPosition(2);
+      writer.integer("A").writeInt(10);
+
+      assertEquals("Child field should be union type", Type.Union, container.getField().getChildren().get(0).getChildren().get(0).getType().getTypeType());
```

Comment:
having the field itself in the message makes it much easier to understand the failure:
Field child = container.getField().getChildren().get(0).getChildren().get(0)
assertEquals("Child field should be union type: " + child, Type.Union, child.getType().getTypeType());